cache/src/main/java/access/DBAccessor.java

```diff
@@ -18,11 +18,11 @@
         return bookRepo.findAll();
     }
 
-    private static final String CACHE_FILE_SAVE_LOCATION = "../../web/cache.txt";
+    private static final String CACHE_FILE_SAVE_LOCATION = "../../web/";
 
-    public void createCache(List<Book> list) {
+    public void createCache(List<Book> list, String libName) {
         try {
-            FileOutputStream outputStream = new FileOutputStream(CACHE_FILE_SAVE_LOCATION);
+            FileOutputStream outputStream = new FileOutputStream(CACHE_FILE_SAVE_LOCATION+libName+"cache.txt");
```

Comment:
Same as above plus one more observation.
You save objects to file, so better extension will be *.bin or leave file without extension.